### PR TITLE
Add autoCloseMenuOnOptionSelection props for auto closing the menu

### LINKED ISSUE #93: Feature: Auto Close Menu
## Type of Feature

- [X] API Extension
- [X] UI Feature
- [ ] other:

## How would it look like

When selecting an input or output option the menu should self-close by default.

Alternatively, an option can be passed to the component that enables or disables this option. This option should be called `autoCloseMenuOnOptionSelection`.

## Resources

* -/-

## Files involved

- N/A

### PR DESCRIPTION
**Pull request by binkpitch**
closed #93 

This PR adds "autoCloseMenuOnOptionSelection" props to support auto closing the menu.

### CODE REVIEW

**Comment by binkpitch** [comment]
@igeligel There are errors from DeepSource that are unrelated to the issue (#93). Should I fix those as well?

**Comment by igeligel** [comment]
No do not have to fix them :)

File: README.md [hunk 1]
```diff
@@ -65,6 +65,7 @@ Some of the unique features this component offers include:
 | onOutOptionsUpdate | function | ✔️ | Updated with new options as the parameter when outOptions language clicked |
 | maxContentLength | number | ❌  | Value that defines the maximum number of characters allowed in the text area. |
 | maxContentLengthIndicator | Object | ❌ | An Object describing how the length indicator is shown. |
+| autoCloseMenuOnOptionSelection | boolean | ❌ | Boolean that defines whether a menu option should self-close after selection.
```
Comment (igeligel) [inline README.md:68]: We are missing the closing `|` here.

```suggestion
| autoCloseMenuOnOptionSelection | boolean | ❌ | Boolean that defines whether a menu option should self-close after selection. |
```

File: src/OptionsOverlay.tsx [hunk 2]
```diff
@@ -83,6 +85,7 @@ export const OptionsOverlay = <T extends Options>(
                 })),
               ];
               onAllMenuOptionsUpdate(updatedOptions as T);
+              onOptionClick();
```
Comment (igeligel) [inline src/OptionsOverlay.tsx:88]: What do you think of returning the option here? Might be useful right?

**Review by igeligel — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (binkpitch) [reply README.md]: Fixed!

File: src/OptionsOverlay.tsx (hunk 2 above)
Comment (binkpitch) [reply src/OptionsOverlay.tsx]: Sound good to me. Will do that.

**Review by igeligel — APPROVED** [review]
Hey @binkpitch. Thanks for your contribution 👍 Would be nice if you could leave a star for the library.
I tagged this Pull Request with the necessary hacktoberfest labels, in case you are participating there. Thanks a lot, contributors like you make Open Source awesome 💯

**Pull request merged**